src/watchdog/__init__.py

```diff
@@ -185,6 +191,42 @@ def get_aws_security_credentials_from_ecs(uri):
     return None
 
 
+def get_aws_security_credentials_from_webidentity(role_arn, token_file):
+    try:
+        with open(token_file, 'r') as f:
+            token = f.read()
+    except Exception as e:
+        if is_fatal:
+            unsuccessful_resp = 'Error reading token file %s: %s' % (token_file, e)
+            fatal_error(unsuccessful_resp, unsuccessful_resp)
+        else:
+            return None, None
+
+    webidentity_url = STS_ENDPOINT_URL + '?' + urlencode({
+        'Version': '2011-06-15',
+        'Action': 'AssumeRoleWithWebIdentity',
+        'RoleArn': role_arn,
+        'RoleSessionName': 'efs-mount-helper',
+        'WebIdentityToken': token
+    })
+
+    unsuccessful_resp = 'Unsuccessful retrieval of AWS security credentials at %s.' % STS_ENDPOINT_URL
+    url_error_msg = 'Unable to reach %s to retrieve AWS security credentials. See %s for more info.' % \
+                    (STS_ENDPOINT_URL, SECURITY_CREDS_WEBIDENTITY_HELP_URL)
+    resp = url_request_helper(webidentity_url, unsuccessful_resp, url_error_msg, headers={'Accept': 'application/json'})
+
+    if resp:
+        creds = resp.get('AssumeRoleWithWebIdentityResponse', {}).get('AssumeRoleWithWebIdentityResult', {}).get('Credentials', {})
```

Comment:
This line is too long.

Reply:
Fixed. I didn't find any style guidelines. Feel free to change it again if you don't like the new format. (Also there are longer lines in the existing code, so this isn't an indication either.)